clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1763,7 +1802,14 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     Expr *ILE = new (SemaRef.getASTContext())
         InitListExpr(SemaRef.getASTContext(), SourceLocation(), ArrayInitExprs,
                      SourceLocation());
-    ILE->setType(FD->getType());
+    QualType ILEType = FD->getType();
+    for (int i = getDims(); i > 1; i--) {
+      const ConstantArrayType *CAT =
+          SemaRef.getASTContext().getAsConstantArrayType(ILEType);
+      assert(CAT && "Should only be called on constant-size array.");
+      ILEType = CAT->getElementType();
+    }
+    ILE->setType(ILEType);
```

Comment:
I don't get what is happening here.

Reply:
Its setting the type of each array correctly using current array depth. E.g. the int[3] of int [2][3] https://godbolt.org/z/nd8WT3. I'll add comments here explaining it.

Reply:
It looks like it is trying to get the element type of the expression?  But it appears to be getting 'int' from 'int[2][3][4][5]', etc, right?

It seems that this would be easier as a ASTContext::getBaseElementType, right?

Reply:
Let me actually confirm this.

Reply:
Elizabeth is right here, it isn't getting the base-element-type, see the comment.